src/ApplicationSampler.cpp

```diff
@@ -371,8 +372,51 @@ namespace geopm
                 process.short_regions.reserve(ApplicationRecordLog::max_region());
             }
         }
+        // Try to pin the sampling thread to a free core
+        std::vector<bool> cpu_enabled(m_num_cpu, false);
+        cpu_enabled.at(sampler_cpu()) = true;
+        std::unique_ptr<cpu_set_t, std::function<void(cpu_set_t *)> > cpu_mask =
+            make_cpu_set(cpu_enabled);
+        (void)sched_setaffinity(0, cpu_set_size(m_num_cpu), cpu_mask.get());
     }
 
+    int ApplicationSamplerImp::sampler_cpu(void)
+    {
+        int result = 0;
+        for (int cpu_idx = m_num_cpu - 1; cpu_idx != -1; --cpu_idx) {
```

Comment:
Could land the controller on hyperthread of application if it is using the last core, but not the hyperthread of the last core.